extmod/modlwip.c

```diff
@@ -1168,16 +1170,17 @@ STATIC mp_uint_t lwip_socket_ioctl(mp_obj_t self_in, mp_uint_t request, uintptr_
             ret |= MP_STREAM_POLL_RD;
         }
 
-        if (flags & MP_STREAM_POLL_WR && tcp_sndbuf(socket->pcb.tcp) > 0) {
+        if (flags & MP_STREAM_POLL_WR && (socket->pcb.tcp == NULL || tcp_sndbuf(socket->pcb.tcp) > 0)) {
             ret |= MP_STREAM_POLL_WR;
         }
 
-        if (socket->state == STATE_PEER_CLOSED) {
-            // Peer-closed socket is both readable and writable: read will
-            // return EOF, write - error. Without this poll will hang on a
-            // socket which was closed by peer.
-            ret |= flags & (MP_STREAM_POLL_RD | MP_STREAM_POLL_WR);
-        }
```

Comment:
Why is this part removed, how is this "peer closed" state handled now?

Reply:
The closed socket is handled with HUP/ERR flag.

Again check this: https://gist.github.com/rkojedzinszky/cbb6710c0cd6b205eda6f32b4bfcd7fc